File: app/src/main/java/com/skyd/imomoe/util/dlna/dms/IResourceServerFactory.java
package com.skyd.imomoe.util.dlna.dms;

interface IResourceServerFactory {
    int getPort();

    IResourceServer getInstance();

    // ----------------------------------------------------------------------------
    // ---- implement
    // ----------------------------------------------------------------------------
    final class DefaultResourceServerFactoryImpl implements IResourceServerFactory {
        private final int port;

        public DefaultResourceServerFactoryImpl(int port) {
            this.port = port;
        }

        @Override
        public int getPort() {
            return port;
        }

        @Override
        public IResourceServer getInstance() {
            // TODO:
            // return new JettyHttpServer(port);
            return new NanoHttpServer(port);
        }
    }
}

File: app/src/main/java/com/skyd/imomoe/util/dlna/dms/IResourceServerFactory.kt
package com.skyd.imomoe.util.dlna.dms


internal interface IResourceServerFactory {
    val port: Int
    val instance: IResourceServer

    // ----------------------------------------------------------------------------
    // ---- implement
    // ----------------------------------------------------------------------------
    class DefaultResourceServerFactoryImpl(override val port: Int) : IResourceServerFactory {
        override val instance: IResourceServer
            // TODO:
            // return new JettyHttpServer(port);
            get() = NanoHttpServer(port)
    }
}